### ISSUE TITLE
Watching isn't Working

### ISSUE THREAD

**Issue by ebebbington** [ebebbington/deno-task-runner-v2#2]
**Summary**
I don't know whether this is down to the following:

* Not working in the original repo
* I implemented the bug 
* Original documentation doesn't actually explain well how to add a watch task. For example, it doesn't give a clear explanation or example file that would run a task and watch a file or directory, that works

**Checklist**

- [ ] Fix the watching aspect so that it works
- [ ] Update README with clear and concise documentation
- [ ] Open a pull request

**Notes**
Some things to remember:

* Check the code changes I made in the `deno-watch-v2` repo to see if this might have broken anything
* Check code changes in this repo to see if it might have broken anything
* Otherwise, best we can do is, just trial and error, for example, create a file with the following:
```
import { task } from "https://deno.land/x/task_runner_v2/mod.ts";

task("prepare", "echo preparing...");
task("server", "deno --allow-net ... app.ts");
task("all", "$prepare", "$server");
task("compile", "echo changed", "$all").watchSync("src");
task("dev-server", "echo restarting...", "$server").watch("server");
```
and just try figure out how this could work e.g. Do we only run the `compile` task which would start the server, and watch for changes, and maybe restart the server again if changes are made? This bit isn't clear as it's hard to understand what happens or what task is called when a file changes

**Expectations**
I can only assume as theres minimal documentation for this, but a task should rerun when a file changes

**Actual result**
Nothing happens, nothing restarts, no visible information

**Additional Information**

In a nutshell, no errors show, but nothing happens when file is changed that is included inside the `watchSync(...)` part

**Comment by ebebbington** [comment]
Upon frther investigation, watching 'does' work, but errors in the following scenarios:

(See [this commit](https://github.com/ebebbington/deno-task-runner-v2/commit/049b3a8fc6b9ae2e84729d1a9f31bc6a40aab8df) where the README was updated following the investigation

* The task detects a change and 'restarts', we get the following error:
```
error: Uncaught NotFound: The system cannot find the file specified. (os error 2)
► $deno$/dispatch_json.ts:40:11
    at DenoError ($deno$/errors.ts:20:5)
    at unwrapResponse ($deno$/dispatch_json.ts:40:11)
    at sendSync ($deno$/dispatch_json.ts:67:10)
    at run ($deno$/process.ts:223:15)
    at kill (runner.ts:66:18)
    at close (runner.ts:43:9)
    at closeResouces (runner.ts:220:14)
    at run (runner.ts:209:7)
```

* When the current working directory is the path to watch
`watch('.')`
And errors with the following:
```
NotFound: The system cannot find the path specified. (os error 3)
    at unwrapResponse ($deno$/dispatch_json.ts:40:11)
    at Object.sendSync ($deno$/dispatch_json.ts:67:10)
    at readDirSync ($deno$/read_dir.ts:25:14)
    at collect (mod.ts:312:20)
    at collect (mod.ts:312:7)
    at collect (mod.ts:312:7)
    at Detector.init (mod.ts:169:5)
    at run (mod.ts:141:34)
    at run.next (<anonymous>)
    at AsyncWatcher.run (runner.ts:207:22)
```

The `watchSync` function seemed to do nothing as in my testing, i had duplicate tasks but one used `watch` and the other used `watchSync`. the one with `watchSync` did nothing upon a change made to the watched file

**Comment by ebebbington** [comment]
Ok both `watch` and `watchSync` do 'work' (kind of), but error in the above scenarios. I've updated the readme again so it can be replicated

**Comment by ebebbington** [comment]
To further investigate this issue, we need to look at the `watch` and `watchSync` function, and why they "can't find the path specified"

**Comment by ebebbington** [comment]
Found out where the error is comng from. Its in `runner.ts`, in the `kill` function, the culprit is:
```
const k = Deno.run({
    args: ["kill", `${p.pid}`],
    stdout: "inherit",
    stderr: "inherit"
  });
```

Try creating a file with just that code and specify any number, run it and you'll see the exact same error.

I can only assume at this point that the PID for the process thats being passed in, is wrong

**Comment by ebebbington** [comment]
So I've discovered, it's down to the `p` object that's passed into the `kill` function not having a PID of a running process. For example, the pid passed in that should be the PID for that running process isn't it, running `Get-Process` in powershell shows the PIDS for the running deno processes, but the PID passed in doesn't match anything those.

So I believe the `Deno.run` isn't retreving the correct pid for a Windows process so it never can actually close it

**Comment by ebebbington** [comment]
Made the changes [here](https://github.com/ebebbington/deno-task-runner-v2/commit/3d755a2b96a0f30e6e3494efbab3eb0c1b8a5d9f) to support running on windows, and the `watchSync` does work after this, but using `watch` still wont kill the processes

**Comment by ebebbington** [comment]
Closed as this functionality has been stripped out due to it's problems, and there already being an alternative: denon
